rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/TaskController.java

```diff
@@ -544,17 +546,79 @@ public ResponseEntity<TaskRepresentationModel> terminateTask(@PathVariable Strin
   public ResponseEntity<TaskRepresentationModel> transferTask(
       @PathVariable String taskId,
       @PathVariable String workbasketId,
-      @RequestBody(required = false) Boolean setTransferFlag)
+      @RequestBody(required = false)
+          TransferTaskRepresentationModel transferTaskRepresentationModel)
       throws TaskNotFoundException,
           WorkbasketNotFoundException,
           NotAuthorizedOnWorkbasketException,
           InvalidTaskStateException {
-    Task updatedTask =
-        taskService.transfer(taskId, workbasketId, setTransferFlag == null || setTransferFlag);
+    Task updatedTask;
+    if (transferTaskRepresentationModel == null) {
+      updatedTask = taskService.transfer(taskId, workbasketId);
+    } else {
+      updatedTask =
+          taskService.transferWithOwner(
+              taskId,
+              workbasketId,
+              transferTaskRepresentationModel.getOwner(),
+              transferTaskRepresentationModel.getSetTransferFlag());
+    }
 
     return ResponseEntity.ok(taskRepresentationModelAssembler.toModel(updatedTask));
   }
 
+  /**
+   * This endpoint transfers a list of Tasks listed in the body to a given Workbasket, if possible.
+   * Tasks transferable without exceptions get transferred independent of other tasks that cannot be
+   * transferred due to errors. If a task cannot be transferred, then no changes to the task will be
+   * made and task remains in old Workbasket. This endpoint requires a body of JSON format that
+   * specifies at least the IDs of tasks to be transferred
+   *
+   * @title Transfer Tasks to another Workbasket
+   * @param workbasketId the Id of the destination Workbasket
+   * @param transferTaskRepresentationModel the TaskIds, owner and setTransferFlag of tasks to be
```

Comment:
"the TaskIds, owner and setTransferFlag of tasks to be  transferred" 
->
"JSON formatted request body containing the TaskIds, owner and setTransferFlag of tasks to be transferred; owner and setTransferFlag are optional, while the TaskIds are mandatory"